opentelemetry-sdk/src/opentelemetry/sdk/util.py

```diff
@@ -26,6 +28,31 @@
     from collections import Sequence
 
 
+def set_timeout_signal_handler():
```

Comment:
This function defines a inner function and then registers it into `signal`, but it is being called in an `__init__`. Keep in mind that you most likely want to do this only once, not every time that an object is initialized. Consider getting rid of this function and just do the registering into `signal` when this module is imported.